src/language-markdown/printer-markdown.js

```diff
@@ -770,43 +770,49 @@ function isPrettierIgnore(node) {
   return match === null ? false : match[1] ? match[1] : "next";
 }
 
+function isInlineNode(node) {
+  return node && INLINE_NODE_TYPES.indexOf(node.type) !== -1;
+}
+
 function shouldNotPrePrintHardline(node, data) {
   const isFirstNode = data.parts.length === 0;
-  const isInlineNode = INLINE_NODE_TYPES.indexOf(node.type) !== -1;
 
   const isInlineHTML =
     node.type === "html" &&
     INLINE_NODE_WRAPPER_TYPES.indexOf(data.parentNode.type) !== -1;
 
-  return isFirstNode || isInlineNode || isInlineHTML;
+  return isFirstNode || isInlineNode(node) || isInlineHTML;
 }
 
-function shouldPrePrintDoubleHardline(node, data) {
-  const isSequence = (data.prevNode && data.prevNode.type) === node.type;
-  const isSiblingNode =
-    isSequence && SIBLING_NODE_TYPES.indexOf(node.type) !== -1;
-
-  const isInTightListItem =
-    data.parentNode.type === "listItem" && !data.parentNode.loose;
+function shouldPrePrintDoubleHardline(node, { parentNode, prevNode }) {
```

Comment:
It'd be better to only do necessary changes or commit the refactoring/bugfix changes separately next time so that it'd be easier to review.